core/include/kernel/user_mode_ctx_struct.h

```diff
@@ -51,6 +54,9 @@ struct user_mode_ctx {
 	bool is_32bit;
 	bool is_initializing;
 	vaddr_t stack_ptr;
+	uint8_t *bbuf;
+	size_t bbuf_size;
+	size_t bbuf_offs;
```

Comment:
maybe a struct for bounce buffer fields.

Reply:
I'm not so keen on that, but I'll add it if you think it's important.